src/rabbit.erl

```diff
@@ -566,75 +614,135 @@ insert_default_data() ->
+  [];
+lager_handlers(tty) ->
+  [{lager_console_backend, debug}];
+lager_handlers({file, FileName}) ->
+  {ok, Dir} = application:get_env(lager, log_root),
+  case rabbit_file:ensure_dir(filename:join(Dir, FileName)) of
+    ok -> ok;
+    {error, Reason} ->
+      throw({error, {cannot_log_to_file, FileName, {cannot_create_parent_dirs, FileName, Reason}}})
+  end,
+  [{lager_file_backend, [
+    {file, FileName}, {level, debug}, {date, ""}, {size, 0}]}].
+
+
+ensure_lager_configured() ->
+    case lager_configured() of
+        false -> configure_lager();
+        true -> ok
     end.
 
-rotate_logs(File, Suffix, Handler) ->
-    rotate_logs(File, Suffix, Handler, Handler).
-
-rotate_logs(undefined, _Suffix, _OldHandler, _NewHandler) -> ok;
-rotate_logs(tty,       _Suffix, _OldHandler, _NewHandler) -> ok;
-rotate_logs(File,       Suffix,  OldHandler,  NewHandler) ->
-    gen_event:swap_handler(error_logger,
-                           {OldHandler, swap},
-                           {NewHandler, {File, Suffix}}).
-
-log_rotation_result({error, MainLogError}, {error, SaslLogError}) ->
-    {error, {{cannot_rotate_main_logs, MainLogError},
-             {cannot_rotate_sasl_logs, SaslLogError}}};
-log_rotation_result({error, MainLogError}, ok) ->
-    {error, {cannot_rotate_main_logs, MainLogError}};
-log_rotation_result(ok, {error, SaslLogError}) ->
-    {error, {cannot_rotate_sasl_logs, SaslLogError}};
-log_rotation_result(ok, ok) ->
-    ok.
+% Lager should have handlers or sinks
+lager_configured() ->
+    application:get_env(lager, handlers) =/= undefined
+    andalso
+    application:get_env(lager, extra_sinks) =/= undefined.
+
+configure_lager() ->
+    application:load(lager),
+    % Turn off reformatting for error_logger messages
+    case application:get_env(lager, error_logger_format_raw) of
+        undefined -> application:set_env(lager, error_logger_format_raw, true);
+        _ -> ok
+    end,
+    case application:get_env(lager, log_root) of
+        undefined -> 
+            application:set_env(lager, log_root, 
+                application:get_env(rabbit, log_base, undefined));
+        _ -> ok
+    end,
+    case application:get_env(lager, handlers) of
+        undefined ->
+            DefaultHandlers = lager_handlers(application:get_env(rabbit, 
+                                                                 error_logger, 
+                                                                 tty)),
+            SaslHandlers = lager_handlers(application:get_env(rabbit, 
+                                                              sasl_error_logger, 
+                                                              tty)),
+            Sinks = [
+                {?LAGER_SINK, [{handlers, DefaultHandlers}]}
+                 % TODO Waiting for PR https://github.com/basho/lager/pull/303
+                 % ,{error_logger_lager_event, [{handlers, SaslHandlers}]}
+                ],
+                Handlers = SaslHandlers,
```

Comment:
what's the purpose of the `Handlers` variables?  Why not just use `SaslHandlers`?